infra/app/api-center-role.bicep

```diff
@@ -0,0 +1,28 @@
+metadata description = 'Creates a role assignment for a service principal.'
+param principalId string
+
+@allowed([
+  'Device'
+  'ForeignGroup'
+  'Group'
+  'ServicePrincipal'
+  'User'
+])
+param principalType string = 'ServicePrincipal'
+param apiCenterName string
+
+var roleDefinitionId = 'b24988ac-6180-42a0-ab88-20f7382dd24c'
+
+resource apiCenter 'Microsoft.ApiCenter/services@2024-03-01' existing = {
+  name: apiCenterName
+}
+
+resource role 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
```

Comment:
We can use infra core roles.

Reply:
It seems we can't pass the `api-center`  to the scope of the role resource through module directly.

Reply:
You should be able to update the role.bicep file from infra/core to accept a resourceId and then find that resource and use it in the scope. https://learn.microsoft.com/en-us/azure/role-based-access-control/role-assignments-template#resource-scope

Reply:
We referenced the following core file and put the `apicenter-access` file in infra/core.
[registry-access.bicep](https://github.com/Azure-Samples/todo-nodejs-mongo/blob/main/infra/core/security/registry-access.bicep)